### ISSUE TITLE
(Bug) Deployment failed if number of reserved addresses more than 45

### ISSUE THREAD

**Issue by dennis00010011b** [poanetwork/token-wizard#775]
Reproducibility 5/5

#### Steps:
1. Start wizard https://wizard.oracles.org/
2. Start creating, add 50 reserved addresses
3. Continue  and finish deployment.

#### Expected result: 
 - token should be deployed without issue
 - user should able to distribute tokens to reserved addresses

#### Actual result:
 - transaction 'Register addresses for reserved tokens' skipped
-  manage page does not contain  button 'Distribute' 

If you are reporting a problem with Token Wizard, please include the following information:

### Which network did you use? (Mainnet, Kovan, Rinkeby, etc.)

Sokol, Rinkeby

### If you were able to create it, what is the URL of your crowdsale?

*https://wizard.oracles.org/invest?addr=0x3cab0D682C7F95Dcad39030895775e7b0799E4c4&networkID=4*

### Do you have screenshots showing the problem?
<img width="1440" alt="screen shot 2018-04-07 at 15 45 40" src="https://user-images.githubusercontent.com/14067096/38461208-19b14272-3a7f-11e8-861a-26a90f2250ed.png">
<img width="1440" alt="screen shot 2018-04-07 at 15 45 35" src="https://user-images.githubusercontent.com/14067096/38461210-1e2c43d8-3a7f-11e8-8167-c94138da8134.png">
<img width="1440" alt="screen shot 2018-04-07 at 15 46 50" src="https://user-images.githubusercontent.com/14067096/38461213-207c7734-3a7f-11e8-87c7-b8ce941c6257.png">
<img width="1440" alt="screen shot 2018-04-07 at 15 51 21" src="https://user-images.githubusercontent.com/14067096/38461215-22b0ac46-3a7f-11e8-9712-225a4694f988.png">
<img width="1440" alt="screen shot 2018-04-07 at 15 53 31" src="https://user-images.githubusercontent.com/14067096/38461216-25286054-3a7f-11e8-9bce-9a70acd39436.png">
<img width="1440" alt="screen shot 2018-04-07 at 15 53 49" src="https://user-images.githubusercontent.com/14067096/38461217-27af919e-3a7f-11e8-9db8-5316e06b18cb.png">
<img width="1440" alt="screen shot 2018-04-07 at 16 00 03" src="https://user-images.githubusercontent.com/14067096/38461218-2a38b95e-3a7f-11e8-9b06-c068600edf8d.png">
<img width="1440" alt="screen shot 2018-04-07 at 16 02 18" src="https://user-images.githubusercontent.com/14067096/38461242-9f949e98-3a7f-11e8-9f96-823a9f9069a9.png">

### Do you see errors in the dev console? If yes, please include a screenshot

Yes
<img width="1440" alt="screen shot 2018-04-07 at 15 53 49" src="https://user-images.githubusercontent.com/14067096/38461220-3b94fd70-3a7f-11e8-985b-f84b9b01d5e8.png">

If you see errors, please right click on them and "Save as..". Zip saved file and attach it to the Issue.

[bundleReservedTokens.log](https://github.com/poanetwork/token-wizard/files/1886798/bundleReservedTokens.log)
[icowizard_Rinkeby_0xE375BB8C92112B841eFD313E7b02ea5b8F3c2aa6.zip](https://github.com/poanetwork/token-wizard/files/1886799/icowizard_Rinkeby_0xE375BB8C92112B841eFD313E7b02ea5b8F3c2aa6.zip)

---

**Comment by fvictorio** [comment]
I'll work on this after #777, so that manual testing it is easier.

**Comment by fvictorio** [comment]
@vbaranov I just realized that this is going to need a change in the contract:

```
  function setReservedTokensListMultiple(
    address[] addrs, 
    uint[] inTokens, 
    uint[] inPercentageUnit, 
    uint[] inPercentageDecimals
  ) public canMint onlyOwner {
    assert(!reservedTokensDestinationsAreSet);
    assert(addrs.length == inTokens.length);
    assert(inTokens.length == inPercentageUnit.length);
    assert(inPercentageUnit.length == inPercentageDecimals.length);
    for (uint iterator = 0; iterator < addrs.length; iterator++) {
      if (addrs[iterator] != address(0)) {
        setReservedTokensList(addrs[iterator], inTokens[iterator], inPercentageUnit[iterator], inPercentageDecimals[iterator]);
      }
    }
    reservedTokensDestinationsAreSet = true;
  }
```

The method does not allow calling it more than once, but if we want to reserve a lot of addresses (say, hundreds) then we'll need to do it in chunks.

I have already made the changes in the frontend to do this, but of course the method fails the second time is called.

But my question is, does it make sense to modify this if we are going to migrate everything to auth_os? How should we approach this?

cc @pablofullana

**Comment by vbaranov** [comment]
Yes, it is contract limitation for now. Let's put this issue on hold until migration to auth_os.
